internal/deckgen/generate.go

```diff
@@ -119,6 +119,19 @@ func ToDeckContent(
 
 	for _, c := range k8sState.Consumers {
 		consumer := file.FConsumer{Consumer: c.Consumer}
+
+		// if a consumer with no username is provided deck wont be able to process it, but we shouldn't
+		// fail the rest of the deckgen either or this will result in one bad consumer being capable of
+		// stopping all updates to the Kong Admin API.
+		if consumer.Username == nil {
+			consumerID := "empty"
+			if consumer.ID != nil {
+				consumerID = *consumer.ID
+			}
+			log.Errorf("invalid consumer received (username was empty), consumer id: %s", consumerID)
```

Comment:
Does this actually have meaningful info in the ID ever? AFAIK we never set IDs for anything and just let Kong auto-populate them:
```suggestion
			log.Errorf("invalid consumer received (username was empty)")
```
If we want to provide meaningful data we'd need to log either the K8S KongConsumer ID or namespace/name, which we've discarded at this point. We may be able to log the CustomID field here, since that's still present, but IMO it's probably not hugely useful--if kubectl lets you filter on CRD fields at all you should be able to run a search for all KongConsumers with an empty Username same as filtering for a specific CustomID.